apps/site/src/app/admin/participants/components/ParticipantsTable.tsx

```diff
@@ -12,22 +18,105 @@ import { Participant } from "@/lib/admin/useParticipants";
 import ParticipantAction from "./ParticipantAction";
 import RoleBadge from "./RoleBadge";
 
+interface EmptyStateProps {
+	title: string;
+	subtitle?: string;
+	action?: ReactElement;
+}
+
 interface ParticipantsTableProps {
 	participants: Participant[];
 	loading: boolean;
 	initiateCheckIn: (participant: Participant) => void;
 	initiatePromotion: (participant: Participant) => void;
 }
 
+const ALL_ROLES = { value: "0", label: "All roles" };
+const ALL_STATUSES = { value: "0", label: "All roles" };
+const SEARCHABLE_COLUMNS = ["_id", "first_name", "last_name", "role", "status"];
+
+function createLabelFunction(columnName: string) {
+	return ({ sorted, descending }: { sorted: boolean; descending: boolean }) => {
+		const sortState = sorted
+			? `sorted ${descending ? "descending" : "ascending"}`
+			: "not sorted";
+		return `${columnName}, ${sortState}.`;
+	};
+}
+
+function EmptyState({ title, subtitle, action }: EmptyStateProps) {
+	return (
+		<Box textAlign="center" color="inherit">
+			<Box variant="strong" textAlign="center" color="inherit">
+				{title}
+			</Box>
+			<Box variant="p" padding={{ bottom: "s" }} color="inherit">
+				{subtitle}
+			</Box>
+			{action}
+		</Box>
+	);
+}
+
 function ParticipantsTable({
 	participants,
 	loading,
 	initiateCheckIn,
 	initiatePromotion,
 }: ParticipantsTableProps) {
-	// TODO: sorting
-	// TODO: search functionality
-	// TODO: role and status filters
+	const [preferences, setPreferences] = useState({
+		pageSize: 20,
+		visibleContent: [
+			"uid",
+			"firstName",
+			"lastName",
+			"role",
+			"status",
+			"action",
+		],
+	});
+	const [filterRole, setFilterRole] = useState(ALL_ROLES);
+	const [filterStatus, setFilterStatus] = useState(ALL_STATUSES);
+	const matchesRole = (p: Participant) => filterRole === ALL_ROLES || p.role === filterRole.label;
+	const matchesStatus = (p: Participant) => filterStatus === ALL_STATUSES || p.status === filterStatus.label;
+
+	const {
+		items,
+		actions,
+		filteredItemsCount,
+		collectionProps,
+		filterProps,
+		paginationProps,
+	} = useCollection(
+		participants,
+		{
+			filtering: {
+				empty: <EmptyState title="No participants" />,
+				noMatch: (
+					<EmptyState
+						title="No matches"
+						action={<Button onClick={() => actions.setFiltering('')}>Clear filter</Button>}
+					/>
+				),
+				filteringFunction: (item, filteringText) => {
+					if (!matchesRole(item)) { return false; }
+					if (!matchesStatus(item)) { return false; }
+					const filteringTextLC = filteringText.toLowerCase();
+
+					return SEARCHABLE_COLUMNS.map(key => item[key]).some(
```

Comment:
Issue: type error with indexing
> Element implicitly has an 'any' type because expression of type 'string' can't be used to index type 'Participant'.
>   No index signature with a parameter of type 'string' was found on type 'Participant'.

Think we'll need to type `SEARCHABLE_COLUMNS` as `(keyof Participant)[]`